velox/exec/StreamingAggregation.cpp

```diff
@@ -239,11 +226,19 @@ void StreamingAggregation::evaluateAggregates() {
     const auto& rows = getSelectivityVector(i);
 
     if (isRawInput(step_)) {
-      aggregate->addRawInput(inputGroups_.data(), rows, args, false);
+      function->addRawInput(inputGroups_.data(), rows, args, false);
     } else {
-      aggregate->addIntermediateResults(inputGroups_.data(), rows, args, false);
+      function->addIntermediateResults(inputGroups_.data(), rows, args, false);
     }
   }
+
+  if (sortedAggregations_) {
+    if (!newGroups.empty()) {
```

Comment:
Let's move this logic out of this method and put it next to function->initializeNewGroups